components/content_settings/core/browser/brave_content_settings_pref_provider.cc

```diff
@@ -22,30 +22,34 @@
 #include "components/prefs/pref_service.h"
 #include "content/public/browser/browser_task_traits.h"
 #include "content/public/browser/browser_thread.h"
+#include "net/base/registry_controlled_domains/registry_controlled_domain.h"
 
 namespace content_settings {
 
 namespace {
 
 Rule CloneRule(const Rule& rule, bool reverse_patterns = false) {
-  auto secondary_pattern = rule.secondary_pattern;
-  if (secondary_pattern ==
+  // brave plugin rules incorrectly use first party url as primary
```

Comment:
it was confusing to flip things back and forth so I reversed them at the start instead and rephrased the comment

Reply:
As for me it became more confusing, because now I need to keep in mind that primary/secondary are probably not what they mean. Maybe renaming to `reversed_primary`/`reversed_secondary` would help

Reply:
actually they are what they mean now, before it was confusing because they were reversed at the end of the method

Reply:
still I would rename them. Maybe `new_primary` and `new_secondary`, or `cloned_primary` etc